ractor.c

```diff
@@ -71,6 +71,7 @@ ractor_lock(rb_ractor_t *r, const char *file, int line)
 
     ASSERT_ractor_unlocking(r);
     rb_native_mutex_lock(&r->sync.lock);
+    r->malloc_gc_disabled = true;
```

Comment:
I don't think `r` is necessarily the current ractor (if it was, we wouldn't need to lock it 😅), so I think that leaves the possibility of a deadlock and creates a race condition